### PR TITLE
Code cells

### PR DESCRIPTION
**Pull request by megan-vo**
* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This is a feature update for the code cells which can now update idyll on shift + enter.

* **What is the current behavior?** (You can also link to an open issue here)
Code cells only execute on close as shown below:

current

* **What is the new behavior (if this is a feature change)?**
Code cells execute on shift + enter and display a warning dialog on close (for unsaved changes).

Shift + enter
shift-enter

Warning
error

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Not that I know of!

* **Other information**:

### CODE REVIEW

File: test/components.test.js [hunk 1]
```diff
@@ -174,3 +175,53 @@ describe('<PropertyList />', () => {
...
+    expect(text).toBe(markup);
+
+    const div = component.find('div');
+    div.simulate('click');
+    expect(onBlur).toHaveBeenCalledTimes(0);
+    div.simulate('blur');
+    expect(onBlur).toHaveBeenCalledTimes(1);
+  });
+
+  it('should display markup and execute on shift + enter', () => {
```
Comment (tanalan) [inline test/components.test.js:206]: Nice tests! Could we also add one for editing the markup (and exiting)?

File: src/render/idyll-display/components/code-cell.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  // Updates markup on blur
+  onBlur = e => {
+    this.toggleEdit(e);
+    this.props.onBlur(this._cellCodeRef.current.textContent);
+  };
+
+  handleKeyDown = e => {
+    e.stopPropagation();
+    if (e.shiftKey && e.keyCode === 13) {
```
Comment (tanalan) [inline src/render/idyll-display/components/code-cell.js:52]: We could also add these to a constants file (keycode constants)

Comment (megan-vo) [reply src/render/idyll-display/components/code-cell.js]: 👍  added

File: test/components.test.js (hunk 1 above)
Comment (megan-vo) [reply test/components.test.js:206]: Made a test for checking content editable 🍣

Comment (tanalan) [reply test/components.test.js:206]: Thank you!

And editing the markup is proving to be quite tricky and difficult! WIP for it atm!

**Comment by mathisonian** [comment]
This is looking great @megan-vo! This seems like a big improvement over the current version. My only question is if it is the expected behavior that the code cell always closes after executing? I think that may be unexpected if a user is trying to test out multiple different parameters for a property.

**Comment by megan-vo** [comment]
Hmm yeah that's a good point @mathisonian! I think what I'll have to do is decouple it from idyll document again like we did with the properties. It currently closes because once it executes it rerenders the entire article again which closes the editing cell. 🤔

**Comment by mathisonian** [comment]
In that case let's maybe save for a separate PR. I've been thinking about other types of cells that we may want to insert, so it might be nice to find a more general solution for this pattern.

**Pull request merged**